<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential expression analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">bootcamp</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>bootcamp/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzvarickabootcamptreeee78f5cb6afb1e03b5c0dee38ca19e9f9bedc864targetblankee78f5ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/zvaricka/bootcamp/tree/ee78f5cb6afb1e03b5c0dee38ca19e9f9bedc864" target="_blank">ee78f5c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomzvarickabootcamptreeee78f5cb6afb1e03b5c0dee38ca19e9f9bedc864targetblankee78f5ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/zvaricka/bootcamp/tree/ee78f5cb6afb1e03b5c0dee38ca19e9f9bedc864" target="_blank">ee78f5c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    bam/
    Ignored:    fastq/
    Ignored:    genome/
    Ignored:    log/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/differential-expression.Rmd</code>) and HTML (<code>docs/differential-expression.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zvaricka/bootcamp/blob/ee78f5cb6afb1e03b5c0dee38ca19e9f9bedc864/analysis/differential-expression.Rmd" target="_blank">ee78f5c</a>
</td>
<td>
Allison Zvarick
</td>
<td>
2020-08-14
</td>
<td>
Publish the DE analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zvaricka/bootcamp/blob/82913326ee34ac38955e531df19850d9c3f7094d/analysis/differential-expression.Rmd" target="_blank">8291332</a>
</td>
<td>
Allison Zvarick
</td>
<td>
2020-08-14
</td>
<td>
Initial commit of RNA-seq scripts.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Load packages.</p>
<pre class="r"><code>library(limma)
library(edgeR)</code></pre>
<p>Import counts.</p>
<pre class="r"><code>rawData &lt;- read.table(&quot;data/counts.txt&quot;,
                      header = TRUE,
                      stringsAsFactors = FALSE)
dim(rawData)</code></pre>
<pre><code>[1] 5292   18</code></pre>
<pre class="r"><code>genes &lt;- subset(rawData, select = Geneid:Length)
counts &lt;- rawData[, 7:18]
rownames(counts) &lt;- genes$Geneid
colnames(counts) &lt;- gsub(&quot;\\.*bam\\.*&quot;, &quot;&quot;, colnames(counts))
group &lt;- c(rep(&quot;mutant&quot;, 6), rep(&quot;wildtype&quot;, 6))
group &lt;- factor(group, levels = c(&quot;wildtype&quot;, &quot;mutant&quot;))
x &lt;- DGEList(counts = counts,
             group = group,
             genes = genes)
class(x)</code></pre>
<pre><code>[1] &quot;DGEList&quot;
attr(,&quot;package&quot;)
[1] &quot;edgeR&quot;</code></pre>
</div>
<div id="process-features" class="section level2">
<h2>Process features</h2>
<!-- Ex 1: log2CPM -->
<p>Calculate log2 counts per million (log2cpm).</p>
<pre class="r"><code>log2cpm &lt;- cpm(x, log = TRUE)</code></pre>
<p>Plot density of expression values for each sample.</p>
<pre class="r"><code>plotDensities(log2cpm, group = group, main = &quot;Raw&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/plotDensities-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Ex 2: Filter -->
<p>Only keep features which have at least 10 counts in at least 4 wildtype or 4 mutant samples.</p>
<pre class="r"><code>keep &lt;- filterByExpr(x, group = group)
sum(keep)</code></pre>
<pre><code>[1] 4904</code></pre>
<pre class="r"><code>x &lt;- x[keep, ]</code></pre>
<p>Re-calculate log2cpm and re-plot densities.</p>
<pre class="r"><code>log2cpm &lt;- cpm(x, log = TRUE)
plotDensities(log2cpm, group = group, main = &quot;Filtered&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/plotDensitiesFiltered-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Ex 3: Normalize -->
<p>Normalize the samples, re-calculate log2cpm, and re-plot densities.</p>
<pre class="r"><code>x &lt;- calcNormFactors(x)
log2cpm &lt;- cpm(x, log = TRUE)
plotDensities(log2cpm, group = group, main = &quot;Normalized&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/normalize-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="process-samples" class="section level2">
<h2>Process samples</h2>
<!-- Ex 1: SNF2 -->
<p>Confirm that the mutant samples are null for SNF2.</p>
<pre class="r"><code>barplot(log2cpm[&quot;YOR290C&quot;, ], main = &quot;SNF2&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/snf2-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Ex 2: PCA -->
<p>Perform PCA.</p>
<pre class="r"><code>plotMDS(log2cpm, gene.selection = &quot;common&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/pca-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove outlier sample.</p>
<pre class="r"><code>x &lt;- x[, colnames(x) != &quot;mutant.06&quot;]
dim(x)</code></pre>
<pre><code>[1] 4904   11</code></pre>
<!-- Ex 3: PCA 2 -->
<p>Re-calculate log2cpm and re-perform PCA.</p>
<pre class="r"><code>log2cpm &lt;- cpm(x, log = TRUE)
plotMDS(log2cpm, gene.selection = &quot;common&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/pca2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="model" class="section level2">
<h2>Model</h2>
<p><span class="math display">\[ Y = \beta_{0} + \beta_{mutant} + \epsilon \]</span></p>
<pre class="r"><code>design &lt;- model.matrix(~x$samples$group)</code></pre>
<pre class="r"><code>v &lt;- voom(x, design, plot = TRUE)</code></pre>
<p><img src="figure/differential-expression.Rmd/voom-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- lmFit(v, design)
fit &lt;- eBayes(fit)</code></pre>
</div>
<div id="explore-results" class="section level2">
<h2>Explore results</h2>
<!-- Ex 1: topTable -->
<p>Count number of differentially expressed features.</p>
<pre class="r"><code>summary(decideTests(fit))</code></pre>
<pre><code>       (Intercept) x$samples$groupmutant
Down             0                  1035
NotSig           0                  3010
Up            4904                   859</code></pre>
<p>View top 10 differentially expressed genes.</p>
<pre class="r"><code>topTable(fit, coef = 2)</code></pre>
<pre><code>         Geneid  Chr  Start    End Strand Length     logFC   AveExpr         t
YML123C YML123C XIII  24037  25800      -   1764 -4.695983  8.083253 -34.40824
YDR033W YDR033W   IV 508147 509109      +    963 -3.672469  8.210557 -33.83913
YGR234W YGR234W  VII 959904 961103      +   1200 -4.237337  7.607611 -34.01898
YER081W YER081W    V 322686 324095      +   1410  3.214874  8.395070  26.55375
YPL019C YPL019C  XVI 514511 517018      -   2508 -2.499383  7.830653 -25.76347
YOR153W YOR153W   XV 619840 624375      +   4536 -2.701209  9.508846 -25.23499
YDR077W YDR077W   IV 600793 601809      +   1017 -3.231475 10.656406 -24.52067
YIL121W YIL121W   IX 132244 133872      +   1629 -2.637367  6.416078 -23.33140
YBR067C YBR067C   II 372103 372735      -    633 -2.925393  8.205030 -22.85657
YOR273C YOR273C   XV 834452 836431      -   1980 -2.113479  7.866356 -22.02893
             P.Value    adj.P.Val        B
YML123C 8.302180e-17 1.780947e-13 28.33391
YDR033W 1.089486e-16 1.780947e-13 28.28710
YGR234W 9.993394e-17 1.780947e-13 28.06935
YER081W 5.572096e-15 6.831390e-12 24.64724
YPL019C 9.075423e-15 8.901175e-12 24.15217
YOR153W 1.267725e-14 1.036154e-11 23.86019
YDR077W 2.013250e-14 1.410426e-11 23.40554
YIL121W 4.476938e-14 2.439434e-11 22.44055
YBR067C 6.226596e-14 3.053522e-11 22.27667
YOR273C 1.124107e-13 5.011473e-11 21.69191</code></pre>
<p>Create a barplot of the top DE feature:</p>
<pre class="r"><code>barplot(log2cpm[&quot;YML123C&quot;, ], las = 2, cex.names = 0.75)</code></pre>
<p><img src="figure/differential-expression.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Ex 2: assessment -->
<p>Visualize p-value distribution.</p>
<pre class="r"><code>hist(fit$p.value[, 2], main = &quot;p-value distribution&quot;)</code></pre>
<p><img src="figure/differential-expression.Rmd/p-values-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Visualize residual variation versus magnitude of expression.</p>
<pre class="r"><code>plotSA(fit)</code></pre>
<p><img src="figure/differential-expression.Rmd/residuals-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- Ex 3: volcano -->
<p>Create a volcano plot.</p>
<pre class="r"><code>volcanoplot(fit, coef = 2, highlight = 5, names = fit$genes$Geneid)</code></pre>
<p><img src="figure/differential-expression.Rmd/volcano-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.10 (Santiago)

Matrix products: default
BLAS:   /gpfs/group/dml129/default/BDR_bootcamp2020/sw/ood_rserver/R/opt/R/4.0.2/lib/R/lib/libRblas.so
LAPACK: /gpfs/group/dml129/default/BDR_bootcamp2020/sw/ood_rserver/R/opt/R/4.0.2/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] edgeR_3.30.3    limma_3.44.3    workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5      knitr_1.29      whisker_0.4     magrittr_1.5   
 [5] lattice_0.20-41 R6_2.4.1        rlang_0.4.7     stringr_1.4.0  
 [9] tools_4.0.2     grid_4.0.2      xfun_0.16       git2r_0.27.1   
[13] htmltools_0.5.0 yaml_2.2.1      digest_0.6.25   rprojroot_1.3-2
[17] later_1.1.0.1   promises_1.1.1  fs_1.5.0        glue_1.4.1     
[21] evaluate_0.14   rmarkdown_2.3   stringi_1.4.6   compiler_4.0.2 
[25] backports_1.1.8 locfit_1.5-9.4  httpuv_1.5.4   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
